packages/block-library/src/template-part/edit/index.js

```diff
@@ -28,6 +28,14 @@ export default function TemplatePartEdit( {
 	// Resolve the post ID if not set, and load its post.
 	const postId = useTemplatePartPost( _postId, slug, theme );
 
+	// TODO: get original slug from meta.
+	const originalSlug = slug;
+	const defaultPostId = useTemplatePartPost( null, originalSlug, theme );
+
+	if ( postId !== defaultPostId && defaultPostId ) {
+		// has customized?
```

Comment:
Theoretically at this point we have enough information to show the revert to default button?